spec/rubocop/cop/rspec/empty_example_group_spec.rb

```diff
@@ -22,15 +22,114 @@
   end
 
   it 'flags an empty top level describe' do
-    expect_offense(<<-RUBY)
+    expect_offense(<<~RUBY)
       describe Foo do
       ^^^^^^^^^^^^ Empty example group detected.
       end
     RUBY
   end
 
+  it 'flags examples improperly nested in invalid scopes' do
+    expect_offense(<<~RUBY)
+      describe Foo do
+      ^^^^^^^^^^^^ Empty example group detected.
+        before do
+          it 'yields a block when given' do
+            value = nil
+
+            helper.feature('whatevs') { value = 5 }
+
+            expect(value).to be 5
+          end
+        end
+      end
+    RUBY
+  end
+
+  it 'ignores example group with examples defined in iterator' do
+    expect_no_offenses(<<~RUBY)
+      describe 'RuboCop monthly' do
+        [1, 2, 3].each do |page|
+          it { expect(newspaper(page)).to have_ads }
+        end
+      end
+    RUBY
+  end
+
+  it 'ignores example group with examples defined in an iterator' do
+    expect_no_offenses(<<~RUBY)
+      describe 'RuboCop weekly' do
+        some_method
+        [1, 2, 3].each do |page|
+          it { expect(newspaper(page)).to have_ads }
+        end
+        more_surroundings
+      end
+    RUBY
+  end
+
+  it 'flags example group with no examples defined in an iterator' do
+    expect_offense(<<~RUBY)
+      describe 'RuboCop Sunday' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^ Empty example group detected.
+        some_method
+        [1, 2, 3].each do |page|
+          no_examples_here
+          and_no_ads_either
+        end
+        more_surroundings
+      end
+    RUBY
+  end
+
+  it 'ignores example group with examples defined in an obscure iterators' do
```

Comment:
Decided not to add this to cop's documentation to avoid proliferating bad practices.